src/babel/transformation/transformers/es6/arrow-functions.js

```diff
@@ -1,4 +1,28 @@
+/**
```

Comment:
These comments probably add more confusion. This isn't strictly what it does and it interacts a lot with `shadow-functions` which isn't mentioned here. No actual transforms on the children are done here.